detekt-rules-complexity/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/ReplaceSafeCallChainWithRun.kt

```diff
@@ -0,0 +1,76 @@
+package io.gitlab.arturbosch.detekt.rules.complexity
+
+import io.gitlab.arturbosch.detekt.api.CodeSmell
+import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.api.Debt
+import io.gitlab.arturbosch.detekt.api.Entity
+import io.gitlab.arturbosch.detekt.api.Issue
+import io.gitlab.arturbosch.detekt.api.Rule
+import io.gitlab.arturbosch.detekt.api.Severity
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.psi.KtElement
+import org.jetbrains.kotlin.psi.KtSafeQualifiedExpression
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.calls.callUtil.getResolvedCall
+import org.jetbrains.kotlin.types.isNullable
+
+/**
+ * Chains of safe calls on non-nullable types are redundant and can be removed by enclosing the redundant safe calls in
+ * a `run {}` block. This improves code coverage and reduces cyclomatic complexity as redundant null checks are removed.
+ *
+ * <noncompliant>
+ * val x = classOrObject.body?.declarations
+ *         ?.filterIsInstance<KtNamedFunction>()
+ *         ?.filter { !isIgnoredFunction(it) }
+ *         ?.size ?: 0
```

Comment:
```suggestion
 * val x = System.getenv()
 *             ?.getValue("HOME")
 *             ?.toLowerCase()
 *             ?.split("/") ?: emptyList()
```

It's a easier example and does not involve detekt specific types/functions.